src/resources/parser/glb-parser.js

```diff
@@ -958,28 +955,18 @@ const createMaterial = function (gltfMaterial, textures, disableFlipV) {
             }
         }
 
-        let scale = uvONE;
-        let offset = uvZERO;
+        const textureTransform = source.extensions?.KHR_texture_transform;
+        if (textureTransform) {
+            const offset = textureTransform.offset || [0, 0];
+            const scale = textureTransform.scale || [1, 1];
+            const rotation = -(textureTransform.rotation || 0);
 
-        const extensions = source.extensions;
-        if (extensions) {
-            const textureTransformData = extensions.KHR_texture_transform;
-            if (textureTransformData) {
-                if (textureTransformData.scale) {
-                    scale = textureTransformData.scale;
-                }
-                if (textureTransformData.offset) {
-                    offset = textureTransformData.offset;
-                }
+            for (map = 0; map < maps.length; ++map) {
+                material[`${maps[map]}MapTiling`] = new Vec2(scale[0], scale[1]);
```

Comment:
I think you mentioned the material just copies values to its internal Vec2's and similar. Maybe we should not new all these Vec2's here, but create Vec2's above where arrays of their values are created?

```
const scale = textureTransform.scale ? new Vec2(textureTransform.scale.x, textureTransform.scale.y) : Vec2.ONE;
...
material[`${maps[map]}MapTiling`] = scale;
```